src/lsm/lsm_cursor.c

```diff
@@ -1576,6 +1649,14 @@ __wt_clsm_open(WT_SESSION_IMPL *session,
 	 */
 	clsm->dsk_gen = 0;
 
+	/* If the next_random option is set, configure a random cursor */
+	WT_ERR(__wt_config_gets_def(session, cfg, "next_random", 0, &cval));
+	if (cval.val != 0) {
+		__wt_cursor_set_notsup(cursor);
+		cursor->next = __clsm_next_random;
+		cursor->reset = __clsm_reset;
```

Comment:
You're right. The notsup function makes the reset a noop, this doesn't matter as reset is redundant on a random cursor.